### DISCUSSION TITLE
Read a raw audio file and send it over WebRTC.

### DISCUSSION THREAD

**Discussion by ranaparti** [devopvoid/webrtc-java#121] [Q&A]
Hi,

I want to read a raw audio file (A/u law) and set it as an audio source. Is there a mechanism to do this with this library?

Thanks,
Ramesh

**Comment by kinsleykajiva** [comment]
tryin reading this  - https://github.com/devopvoid/webrtc-java/discussions/90   if it helps share as well

**Comment by devopvoid** [comment]
You can now achieve that using a `CustomAudioSource`, documented [here](https://jrtc.dev/#/guide/custom_audio_source).
